haystack/tools/parameters_schema_utils.py

```diff
@@ -47,6 +47,42 @@ def _get_param_descriptions(method: Callable) -> Tuple[str, Dict[str, str]]:
     return parsed_doc.short_description or "", param_descriptions
 
 
+def _get_component_param_descriptions(component: Any) -> Tuple[str, Dict[str, str]]:
+    """
+    Get parameter descriptions from a component, handling both regular Components and SuperComponents.
+
+    For regular components, this extracts descriptions from the run method's docstring.
+    For SuperComponents, this extracts descriptions from the underlying pipeline components.
+
+    :param component: The component to extract parameter descriptions from
+    :returns: A tuple of (short_description, param_descriptions)
+    """
+    # Import here to avoid circular imports
+    from haystack.core import SuperComponent
+
+    # Get descriptions from the component's run method
+    short_desc, param_descriptions = _get_param_descriptions(component.run)
```

Comment:
I think we should also fetch the run method description from the underlying components in SuperComponent like we are doing for the parameters. Otherwise we will always get the `_SuperComponent.run` method description which is quite vague 

> Runs the wrapped pipeline with the provided inputs.
Steps:
> 1. Maps the inputs from kwargs to pipeline component inputs
> 2. Runs the pipeline
> 3. Maps the pipeline outputs to the SuperComponent's outputs

Reply:
Could also be good to combine the run descriptions together like in this example https://github.com/deepset-ai/haystack/pull/9403#discussion_r2097139061 if needed